### PR TITLE
chore(build): Rename gulpfile.js to ES6-friendly gulpfile.babel.js

### LINKED ISSUE #57: Switch to ES6-based babel file
It's ES6 all the way down!
We only need to rename the gulpfile.js to gulpfile.babel.js, and then all the ES6 magic should just work.

https://markgoodyear.com/2015/06/using-es6-with-gulp/

### PR DESCRIPTION
**Pull request by pdehaan**
Fixes #57

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,15 +1,16 @@
-tmp/
-.env
-*.sw?
+*.log
```
Comment (pdehaan) [inline .gitignore:3]: This `*.log` line the only added/changed bit in this file. The rest were just me being a jerk and A-Z sorting to make items easier to scan.

**Comment by meandavejustice** [comment]
@pdehaan after the latest merge we got some conflicts

**Comment by pdehaan** [comment]
@meandavejustice Rebased, for your r?ing pleasure.

**Comment by lmorchard** [comment]
Is this PR working for everyone else? I'm getting this error since I did a `git pull` and rebuilt my containers:

```
client_build_1 | 14 Sep 19:47:43 - [nodemon] v1.5.0
client_build_1 | 14 Sep 19:47:43 - [nodemon] to restart at any time, enter `rs`
client_build_1 | 14 Sep 19:47:43 - [nodemon] watching: /app/gulpfile.babel.js
client_build_1 | 14 Sep 19:47:43 - [nodemon] starting `node /home/webdev/node_modules/gulp/bin/gulp.js`
client_build_1 | [19:47:43] Failed to load external module babel-core/register
client_build_1 | [19:47:43] Failed to load external module babel/register
client_build_1 | /app/gulpfile.babel.js:1
client_build_1 | (function (exports, require, module, __filename, __dirname) { import autoprefi
client_build_1 |                                                               ^^^^^^
client_build_1 | SyntaxError: Unexpected reserved word
```

**Comment by pdehaan** [comment]
Ugh, I can repro the error.

I think this is a **nodemon** issue. https://babeljs.io/docs/setup/#nodemon

https://github.com/mozilla/idea-town/blob/master/bin/run-client-build.sh would also need to be tweaked to call a new npm script for **babel-node**.

Or, just revert the gulpfile.js since I don't know if full ES6 support is a big enough win to complicate the build process further.

**Comment by lmorchard** [comment]
FWIW, this is also happening for me without nodemon:

```
➜  idea-town git:(master) docker exec -t -i ideatown_client_build_1 bash
webdev@e486b609ca54:/app$ NODE_PATH=/home/webdev/node_modules/ /usr/local/bin/node /home/webdev/node_modules/gulp/bin/gulp.js
[20:00:58] Failed to load external module babel-core/register
[20:00:58] Failed to load external module babel/register
```

**Comment by pdehaan** [comment]
@lmorchard Let's revert this and stick w/ vanilla ES5 for the gulpfile.

**Comment by pdehaan** [comment]
Weird, I can definitely see the error on the Docker machine, but it seems to work locally for me:

``` sh
$ ./node_modules/.bin/nodemon --watch gulpfile.babel.js ./node_modules/gulp/bin/gulp.js
15 Sep 11:15:07 - [nodemon] v1.5.0
15 Sep 11:15:07 - [nodemon] to restart at any time, enter `rs`
15 Sep 11:15:07 - [nodemon] watching: gulpfile.babel.js
15 Sep 11:15:07 - [nodemon] starting `node ./node_modules/gulp/bin/gulp.js`

[11:15:08] Failed to load external module babel-core/register
[11:15:08] Requiring external module babel/register
[11:15:10] Using gulpfile ~/dev/github/idea-town/gulpfile.babel.js

[11:15:10] Starting 'build'...
[11:15:10] Starting 'clean'...
[11:15:10] Finished 'clean' after 12 ms
[11:15:10] Starting 'vendor'...
[11:15:10] Starting 'npm:tabzilla:img'...
[11:15:10] Starting 'npm:tabzilla:css'...
[11:15:10] Starting 'npm:normalize'...
[11:15:10] Finished 'npm:tabzilla:css' after 15 ms
[11:15:10] Finished 'npm:normalize' after 12 ms
[11:15:10] Finished 'npm:tabzilla:img' after 22 ms
[11:15:10] Finished 'vendor' after 23 ms
[11:15:10] Starting 'scripts'...
[11:15:13] Finished 'scripts' after 3.07 s
[11:15:13] Starting 'styles'...
[11:15:13] Finished 'styles' after 139 ms
[11:15:13] Starting 'images'...
[11:15:13] Finished 'images' after 13 ms
[11:15:13] Finished 'build' after 3.26 s
[11:15:13] Starting 'watch'...
[11:15:13] Finished 'watch' after 17 ms
[11:15:13] Starting 'default'...
[11:15:13] Finished 'default' after 5.76 μs
```

**Pull request merged**